tests/notice_xml_tests.py

```diff
@@ -129,3 +129,296 @@ def test_source_is_local(self):
+                u'raw_name': u'DEPARTMENT OF JUSTICE',
+                u'json_url': u'%s%s' % (u'https://www.federalregister.gov/',
+                                        u'api/v1/agencies/268.json'),
+                u'id': 268
+            },
+            {
+                u'name': u'ATF subsubagency',
+                u'parent_id': 100023,
+                u'url': u'%s%s' % (u'https://www.federalregister.gov/',
+                                   u'agencies/atf-subsubagency'),
+                u'raw_name': u'SUBSUBAGENCY OF ATF',
+                u'json_url': u'%s%s' % (u'https://www.federalregister.gov/',
+                                        u'api/v1/agencies/100072.json'),
+                u'id': 100072
+            },
+        ]
+        with XMLBuilder("ROOT") as ctx:
+            with ctx.DATES():
+                ctx.P("Effective on May 4, 2004")
+        xml = notice_xml.NoticeXML(ctx.xml)
+        xml.derive_agencies(agencies=agencies_info)
+        self.assertEquals(len(xml.xpath("//EREGS_AGENCIES")), 1)
+        eregs_agencies = xml.xpath("//EREGS_AGENCIES")[0]
+        self.assertEquals(len(eregs_agencies.xpath("//EREGS_AGENCY")), 1)
+        doj = eregs_agencies.xpath("//EREGS_AGENCY")[0]
+        self.assertEquals(doj.attrib["eregs-agency-name"],
+                          "Justice Department")
+        self.assertEquals(doj.attrib["eregs-agency-raw-name"],
+                          "DEPARTMENT OF JUSTICE")
+        self.assertEquals(doj.attrib["eregs-agency-id"], "268")
+        self.assertEquals(len(doj.xpath("//EREGS_SUBAGENCY")), 3)
+        self.assertEquals(len(doj.xpath("EREGS_SUBAGENCY")), 1)
+        atf = doj.xpath("//EREGS_SUBAGENCY")[0]
+        self.assertEquals(atf.attrib["eregs-agency-name"],
+                          u'Alcohol, Tobacco, Firearms, and Explosives Bureau')
+        self.assertEquals(atf.attrib["eregs-agency-raw-name"],
+                          "%s%s" % ("Bureau of Alcohol, Tobacco, ",
+                                    "Firearms and Explosives"))
+        self.assertEquals(atf.attrib["eregs-agency-id"],
+                          "19")
+        self.assertEquals(len(atf.xpath("EREGS_SUBAGENCY")), 1)
+        subatf = atf.xpath("EREGS_SUBAGENCY")[0]
+        self.assertEquals(subatf.attrib["eregs-agency-name"],
+                          u'ATF subagency')
+        self.assertEquals(subatf.attrib["eregs-agency-raw-name"],
+                          u"SUBAGENCY OF ATF")
+        self.assertEquals(subatf.attrib["eregs-agency-id"],
+                          u"100023")
+        subsubatf = subatf.xpath("EREGS_SUBAGENCY")[0]
+        self.assertEquals(subsubatf.attrib["eregs-agency-name"],
+                          u'ATF subsubagency')
+        self.assertEquals(subsubatf.attrib["eregs-agency-raw-name"],
+                          u"SUBSUBAGENCY OF ATF")
+        self.assertEquals(subsubatf.attrib["eregs-agency-id"],
+                          u"100072")
```

Comment:
It looks like there's a lot of redundancy in these tests. Is there anything that could be abstracted out?

I'd recommend using the XML builder for verification. For example, you could write:
```python
actual = xml.xpath("//EREGS_AGENCIES")[0]
with XMLBuilder("EREGS_AGENCIES") as ctx:
    with ctx.EREGS_AGENCY(name='Justice Department', raw='DEPARTMENT OF JUSTICE', id=268):
        ctx.EREGS_SUBAGENCY(name='Alcohol, Tobacco, Firearms, and Explosives Bureau', ...
...
self.assertEqual(etree.tostring(actual), ctx.xml_str)
```
which feels a tad easier to read.